Assets/Scripts/Layers/LayerTypes/ReferencedLayer.cs

```diff
@@ -1,46 +1,83 @@
 using System;
-using System.Collections;
-using System.Collections.Generic;
 using Netherlands3D.Twin.Projects;
-using Netherlands3D.Twin.UI.LayerInspector;
 using UnityEngine;
 using UnityEngine.Events;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
 
-namespace Netherlands3D.Twin
+namespace Netherlands3D.Twin.Layers
 {
     public abstract class ReferencedLayer : MonoBehaviour
     {
+        [SerializeField] private string prefabIdentifier;
+        public string PrefabIdentifier => prefabIdentifier;
```

Comment:
Can we prevent storing the prefab identifier in the data? Prefabs are application-specific and not a fixed datapoint, if the application changes; future projects will break. And how does this work in the build? The reliance on editor / onValidate does not work in a build

Reply:
I need to store a reference to the prefab that has to be instantiated, I used the prefabID so that it would be the same as the meta id of the prefab when developing, I could have also used a new UUID, but I thought this would be more useful

Reply:
I'll add a tech debt ticket to discuss this in more detail; because I think it would be better to store a reference to the type of subject being stored and have the application figure out which prefab to use for that subject. This will make the loading and saving more flexible and less tied to Unity and prone to breaking if we change prefabs

Reply:
For the record: I just talked this over with Tom; and it is limited understanding from my part. The Prefab id is only used as an initializer and not for the future. It may mean that we might need to rename this field to clarify this